core/OpenStore365.Serverless.Auth/AAD_B2C/IoC/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,10 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+
+namespace OpenStore365.Serverless.Auth.AAD_B2C.IoC
+{
+    public static class IServiceCollectionExtensions
+    {
+        public static IServiceCollection Add_AAD_B2C_HttpHeaderUserProvider(this IServiceCollection services) =>
+            services.AddSingleton<IUserProvider, HttpHeaderUserProvider>();
```

Comment:
Explanation on Singleton here please, isn't User scoped?

Reply:
It's an object that takes the `HttpRequest` and extracts HTTP header values regarding user data into a new `User` object. Making it a singleton we have guarantee that only one object will be created per Azure Function instance in its memory. If we would create an instance of it each time a new HTTP request arrives we would be wasting memory for no reason.